File: src/main/java/net/softbell/bsh/dto/view/admin/CenterSettingSummaryCardDto.java
package net.softbell.bsh.dto.view.admin;

import lombok.Getter;
import lombok.Setter;
import net.softbell.bsh.domain.entity.CenterSetting;

/**
 * @Author : Bell(bell@softbell.net)
 * @Description : 센터 설정 카드정보 DTO
 */
@Getter
@Setter
public class CenterSettingSummaryCardDto
{
	private Byte isEnabled;
	private Byte iotAction;
	private Byte iotControl;
	private Byte iotMonitor;
	private Byte iotNode;
	private Byte iotReserv;
	private Byte iotTrigger;
	private Byte webAuthMode;
	private Integer webLoginFailBanTime;
	private Integer webLoginFailCheckTime;
	private Byte webLoginFailMaxCount;
	private Byte webMaintenance;
	private Byte webRegister;
	
	public CenterSettingSummaryCardDto(CenterSetting entity)
	{
		// Exception
		if (entity == null)
			return;
		
		// Convert
		this.isEnabled = entity.getIsEnabled();
		this.iotAction = entity.getIotAction();
		this.iotControl = entity.getIotControl();
		this.iotMonitor = entity.getIotMonitor();
		this.iotNode = entity.getIotNode();
		this.iotReserv = entity.getIotReserv();
		this.iotTrigger = entity.getIotTrigger();
		this.webAuthMode = entity.getWebAuthMode();
		this.webLoginFailBanTime = entity.getWebLoginFailBanTime();
		this.webLoginFailCheckTime = entity.getWebLoginFailCheckTime();
		this.webLoginFailMaxCount = entity.getWebLoginFailMaxCount();
		this.webMaintenance = entity.getWebMaintenance();
		this.webRegister = entity.getWebRegister();
	}
}

File: src/main/kotlin/net/softbell/bsh/dto/view/admin/CenterSettingSummaryCardDto.kt
package net.softbell.bsh.dto.view.admin

import net.softbell.bsh.domain.entity.CenterSetting

/**
 * @author : Bell(bell@softbell.net)
 * @description : 센터 설정 카드정보 DTO
 */
class CenterSettingSummaryCardDto(entity: CenterSetting) {
    val enabled: Byte = entity.isEnabled
    val iotAction: Byte = entity.iotAction
    val iotControl: Byte = entity.iotControl
    val iotMonitor: Byte = entity.iotMonitor
    val iotNode: Byte = entity.iotNode
    val iotReserv: Byte = entity.iotReserv
    val iotTrigger: Byte = entity.iotTrigger
    val webAuthMode: Byte = entity.webAuthMode
    val webLoginFailBanTime: Int = entity.webLoginFailBanTime
    val webLoginFailCheckTime: Int = entity.webLoginFailCheckTime
    val webLoginFailMaxCount: Byte = entity.webLoginFailMaxCount
    val webMaintenance: Byte = entity.webMaintenance
    val webRegister: Byte = entity.webRegister
}